google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -799,13 +803,25 @@ public LifecycleCondition build() {
     }
 
     /**
-     * Base class for the Action to take when a Lifecycle Condition is met. Specific Actions are
+     * Base class for the Action to take when a Lifecycle Condition is met. Supported Actions are
      * expressed as subclasses of this class, accessed by static factory methods.
      */
-    public abstract static class LifecycleAction implements Serializable {
+    public static class LifecycleAction implements Serializable {
```

Comment:
Abstract classes can't be instantiated so we do if we want to allow for unsupported action types to be created in this way.

The alternative would be to make an implementation of LifecycleAction for that purpose, like "CustomLifecycleAction" or something, but I'm worried that doing that would be confusing, or imply that this is the correct way to use action types that aren't in the library (when in actuality users should upgrade to the latest version of the library)

Reply:
Ah, totally missed that. Thanks Jesse, I would confirm what happens when a say setStorageClass doesn't have storageClass in the rule (the case an action type is added that requires additional parameters) and see how GCS handles the request on a PATCH request. 

I suspect it will fail; so we may want to clarify if we want the library *not* to create an instance for that unknown action type.